### PR TITLE
Feature Branch

### PR DESCRIPTION
**Pull request by rsundar**
This project is incomplete but the core functionality is done

# FEATURES CREATED

* Form where user can enter their city for which they wish to check the weather.
* Can pull weather data from openweathermap api.

# TODO

* Add a geolocation api to automatically detect the location.
* Create the UI using bulma
* Create a chart of the hourly forecast for the next 24 hours.

### CODE REVIEW

File: src/modules/fetch.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import { getData } from './dailyData.js'
+import getData from './dailyData';
 
 const apiKey = '356c3c6429275ec5cc808b231f9c0064';
 const url = 'https://api.openweathermap.org/data/2.5/weather?';
 
-async function retrieveData(city, metric) {
+export default async function retrieveData(city, metric) {
```
Comment (lucasmazo23) [inline src/modules/fetch.js]: image

File: src/modules/index.js [hunk 2]
```diff
@@ -1,22 +1,9 @@
...
-  }
-  else {
-    metric = "metric";
-  }
-  let weather = retrieveData(city, metric);
-  weather.then(value => showData(value));
-}
+import { retrieveData } from './fetch.js'
 
 function showData(weatherData) {
```
Comment (lucasmazo23) [inline src/modules/index.js]: image

File: src/modules/index.js [hunk 3]
```diff
@@ -27,3 +14,17 @@ function showData(weatherData) {
     `;
   ul.insertAdjacentHTML('beforeend', table);
 }
+
+function submitForm() {
+  const city = document.forms.cityForm.city.value;
+  let metric = '';
+  if (document.forms.cityForm.i.checked) {
```
Comment (lucasmazo23) [inline src/modules/index.js:28]: [OPTIONAL] Consider using object destructuring

image

**Review by lucasmazo23 — CHANGES_REQUESTED** [review]
## Changes required 🔧
### You are doing a great job so far, and the project is looking good.

### For the project to get approved kindly implement these changes:

### Required changes:
- [x] [REQUIRED] The app should change its appearance based on the results from the search
image
[source](https://www.theodinproject.com/courses/javascript/lessons/weather-app)
- [x] [REQUIRED] Try to use ES6 suntax when possible, `const myFunc = () => {}` instead of `function myFunc(){}` (examples in my comments below)
image
[Source](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/)

### Happy coding! 💻
## Lucas Mazo

File: src/modules/index.js (hunk 2 above)
Comment (rsundar) [reply src/modules/index.js]: done

File: src/modules/fetch.js (hunk 1 above)
Comment (rsundar) [reply src/modules/fetch.js]: done

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @rsundar 
### You have done well so far :clap: However, you have a few issues to fix:-
- [x] Fix the conflict in the pull request, see the image below for the specific file
conflict
### Make the change and submit for another code review. Good luck.
- Happy coding!

**Comment by rsundar** [comment]
# CHANGES MADE
- Made the changes as required by @lucasmazo23 and @rOluochKe 
- The app changes the picture depending on the weather condition and city.

I will add the charting facility of the hourly forecast later as an improvement while I make the portfolio..

**Review by trekab — CHANGES_REQUESTED** [review]
### Changes Required! :recycle: 
Hi  @rsundar! Your effort to work on the feedback from the last reviews is commendable! :+1: There are a few details to work on to complete this project though:
- [x] Linters config: Your linter checks are not running. Please reconfigure linting for your repository. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/)). Make use of the [config files](https://github.com/microverseinc/linters-config/tree/master/javascript) prepared for this section.
- [x] Nice job implementing a descriptive README. However, your live demo link doesn't lead to a deployed version of your app. Please fix this.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[Portfolio](https://trekab.github.io/portfolio/) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

**Review by Dandush03 — APPROVED** [review]
# Status :rocket: Approved
# Great Work Implementing Necessary Changes
# Happy Coding :computer:

**Pull request merged**